src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -0,0 +1,204 @@
+
+/*!
+ * \ingroup DomainCreatorsGroup
+ *
+ * \brief A Domain for testing purposes.
+ *
+ * \image html binary_compact_object_domain.png "A BHNS domain."
+ *
+ * Creates a 3D Domain consisting of four/five layers of blocks:
+ * - 0: The block at the center of the compact object, if not excised.
+ * - 1: The first layer, has the blocks for the compact object.
+ * - 2: The second layer, has the blocks that surround the object with a cube.
+ * - 3: The third layer, has the blocks that surround each cube with a
+ * half-cube.
+ * - 4: The fourth layer is made of the blocks that form the outer sphere.
+ *
+ * `ObjectA` and `ObjectB` refer to the Shell or Sphere making up each compact
+ * object.
+ * `enveloping cube` and `enveloping sphere` refer to the bounding surfaces of
+ * the outer Sphere of ten wedges making up the last layer of the domain.
+ *
+ * The plane along which the domain divides into two hemispheres is referred
+ * to as the "cutting plane".
+ */
+
+template <typename TargetFrame>
+class BinaryCompactObject : public DomainCreator<3, TargetFrame> {
+ public:
+  struct InnerRadiusObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "Coordinate radius of this compact object."};
+  };
+
+  struct OuterRadiusObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "Coordinate radius of this compact object."};
+  };
+
+  struct XCoordObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "x-coordinate of this compact object."};
+  };
+
+  struct ExciseInteriorA {
+    using type = bool;
+    static constexpr OptionString help = {
+        "Set to `true` for a BH, `false` otherwise."};
+  };
+
+  struct InnerRadiusObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "Areal radius of this compact object."};
+  };
+
+  struct OuterRadiusObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "Areal radius of this compact object."};
+  };
+
+  struct XCoordObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "x-coordinate of this compact object."};
+  };
+
+  struct ExciseInteriorB {
+    using type = bool;
+    static constexpr OptionString help = {
+        "Set to `true` for a BH, `false` otherwise."};
+  };
+
+  struct RadiusOuterCube {
+    using type = double;
+    static constexpr OptionString help = {
+        "Radius circumscribing the layer of Frustums."};
+  };
+
+  struct RadiusOuterSphere {
+    using type = double;
+    static constexpr OptionString help = {"Radius of the entire domain."};
+  };
+
+  struct InitialRefinement {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Initial refinement level in each dimension."};
+  };
+
+  struct InitialGridPoints {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Initial number of grid points in each dimension."};
```

Comment:
is this per spectral element?   "Initial number of grid-points in each dimension per element"